src/lib/event.ts

```diff
@@ -55,19 +54,29 @@ class EventSystem<EventInterface extends Record<keyof EventInterface, Callback>>
 	}
 
 	fire<T extends keyof EventInterface>(eventName: T, ...args: Parameters<EventInterface[T]>): void {
-		const event = this.events.get(eventName);
-		if (event === undefined) return print(`No event listeners for event ${eventName}`); // TODO: Switch to Tina's Logger
+		const event = this.events.get(eventName as string);
+		if (event === undefined) return print(`No event listeners for event ${eventName as string}`); // TODO: Switch to Tina's Logger
 		
 		event._callbacks.forEach((callback) => {
 			callback(...args);
 		});
 
-		if (event._waitingThreads.size() > 0) {
-			event._waitingThreads.forEach((thread) => {
-				coroutine.resume(thread);
-			});
+			if (event._waitingThreads.size() >= 0) {
+				event._waitingThreads.forEach((thread) => {
+					coroutine.resume(thread, ...args);
+				});
+			}
```

Comment:
Shouldn't we remove the waiting thread when it stops waiting?

Reply:
Yes